### PR TITLE
updated with useCennznet to connect to extension and return all accounts and api instance

### LINKED ISSUE #324: React hook support + extension integration
Provide a one-liner for dapp developers to connect with extension get available accounts and API session object

can probably go into its own package `@cennznet/api-react`

Aim is to reduce boiler plate for dapp developers
https://github.com/cennznet/cennzx-ui/blob/047a1e88d5841289696890f6f32e7aee62ad15aa/src/redux/reducers/extension.reducer.ts#L6-L55

```js
function App() {
    // connect api object
    // connect with extension
    // get extension addresses
    const {  cennznet, accounts } = useCENNZ('My App', { network: 'rata' });
    return <>
}
```

suggested by @miketamis

### PR DESCRIPTION
**Pull request by eupston**
closes #324

Will ideally be using this to refactor NFT Demo

### CODE REVIEW

File: packages/api/src/util/useCennznet.ts
Comment (holygits) [inline packages/api/src/util/useCennznet.ts:39]: we should allow the extension to work with different networks at different times.

similar to what we did in cennzx something like 
`cennznet-ext-meta-{api.runtimeVersion.specName}-{api.runtimeVersion.specVersion}`
should be the key.

**Review by holygits — COMMENTED** [review]
nice!

**Comment by eupston** [comment]
Have tested in NFT Demo and works well.

Currently if the extension isn't installed `accounts` returns null, and if it is installed but no accounts are present than it returns an empty array. We may want to update the user docs at some point to highlight this detail, so devs can handle all situations appropriately on frontend.

File: packages/api/src/util/useCennznet.ts
Comment (eupston) [reply packages/api/src/util/useCennznet.ts]: sweet yeah good call, updated now. 
Decided to go with network name as key

File: packages/api/src/hooks/UseCennznet.ts
Comment (KarishmaBothara) [inline packages/api/src/hooks/UseCennznet.ts:21]: will this only provide the promise version of API and not rx version?

File: packages/api/src/util/cennznetExtensions.ts
Comment (KarishmaBothara) [inline packages/api/src/util/cennznetExtensions.ts:1]: I think we can reuse this type from 
https://github.com/cennznet/api.js/blob/master/packages/types/src/interfaces/extrinsic/signedExtensions/cennznet.ts

File: packages/api/src/util/useCennznet.ts
Comment (KarishmaBothara) [reply packages/api/src/util/useCennznet.ts]: I think better to go with specVersion as for the same network, change in specVersion will introduce new metadata..

Comment (eupston) [reply packages/api/src/util/useCennznet.ts]: sweet updated now

File: packages/api/src/hooks/UseCennznet.ts
Comment (eupston) [reply packages/api/src/hooks/UseCennznet.ts:23]: added option to return rx api as well

File: packages/api/src/util/cennznetExtensions.ts
Comment (eupston) [reply packages/api/src/util/cennznetExtensions.ts]: o cool yup changed to use this type instead

File: packages/api/src/hooks/UseCennznet.ts
Comment (holygits) [reply packages/api/src/hooks/UseCennznet.ts:23]: consider using another function instead of bool
e.g.
```ts
UseCennznet
UseCennznetRx
```

**Review by holygits — APPROVED** [review]
lgtm!
my only reservation is we should remove the metadata management stuff from here once extension PR
https://github.com/cennznet/extension/pull/9 lands and is published

File: packages/api/src/hooks/UseCennznet.ts
Comment (eupston) [reply packages/api/src/hooks/UseCennznet.ts:23]: hmm yeah, probably more inline with our current API

**Comment by eupston** [comment]
> lgtm!
> my only reservation is we should remove the metadata management stuff from here once extension PR
> [cennznet/extension#9](https://github.com/cennznet/extension/pull/9) lands and is published

yeah do you reckon we should wait until that PR is merged? or do it after the fact?

**Comment by holygits** [comment]
I think we can merge this but wouldn't put it into an official release until metadata management is moved to extension, no harm in doing a pre-release with this change for testing purposes

File: packages/api/src/hooks/UseCennznet.ts
Comment (eupston) [reply packages/api/src/hooks/UseCennznet.ts:23]: cool updated now

File: packages/api/src/hooks/UseCennznet.ts
Comment (KarishmaBothara) [inline packages/api/src/hooks/UseCennznet.ts:88]: can remove this delete specific class template with 
```

// remove all the classes from the spectypes
// the metadata that gets updated on the cennznet extension requires types of type (Record<string, string|object>)
                        const filteredSpecTypes = Object.keys(specTypes)
                            .filter(key => {
                                if (typeof specTypes[key] === 'function') {
                                    return false;
                                }
                                return true;
                            })
                            .reduce((obj, key) => {
                                obj[key] = specTypes[key];
                                return obj;
                            }, {});
```

Comment (KarishmaBothara) [reply packages/api/src/hooks/UseCennznet.ts:88]: which would mean, in future even if we add new classes we won't need to add a delete that particular class line here..
The code will still work...

Comment (eupston) [reply packages/api/src/hooks/UseCennznet.ts]: oh yup cool just updated

**Review by KarishmaBothara — APPROVED** [review]
lgtm

**Pull request merged**
